source/compiler-core/slang-diagnostic-sink.h

```diff
@@ -238,10 +249,17 @@ class DiagnosticSink
 
 protected:
     void diagnoseImpl(SourceLoc const& pos, DiagnosticInfo const& info, int argCount, DiagnosticArg const* const* args);
+    void diagnoseImpl(DiagnosticInfo const& info, const UnownedStringSlice& formattedMessage);
+
+        /// If set all diagnostics (as formatted by *this* sink, will be routed to the parent).
```

Comment:
Seems like the parenthetical here should end earlier:
```
/// If set all diagnostics (as formatted by *this* sink), will be routed to the parent.
```

Reply:
It's not something we should do write away, but this almost suggests that we'd eventually want some distinction between:

* The interface that clients talk to when they want to emit diagnostics, which handles formatting policy
* The interface that handles actually writing diagnostics (already formatted) to a buffer/console/whatever